PartitionPtscotch.h

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ *  This file is part of PUML
+ *
+ *  For conditions of distribution and use, please see the copyright
+ *  notice in the file 'COPYING' at the root directory of this package
+ *  and the copyright notice at https://github.com/TUM-I5/PUMGen
+ *
+ * @copyright 2019 Technische Universitaet Muenchen
+ * @author David Schneller <david.schneller@tum.de>
+ */
+
+#ifndef PUML_PARTITIONPTSCOTCH_H
+#define PUML_PARTITIONPTSCOTCH_H
+
+#ifdef USE_MPI
+#include <mpi.h>
+#endif // USE_MPI
+
+#ifndef USE_PTSCOTCH
+#warning PTSCOTCH is not enabled.
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <ptscotch.h>
+#include <cmath>
+
+#include "PartitionBase.h"
+#include "PartitionGraph.h"
+#include "Topology.h"
+
+namespace PUML
+{
+
+template<TopoType Topo>
+class PartitionPtscotch : public PartitionBase<Topo>
+{
+
+public:
+	PartitionPtscotch(int mode) : mode(mode) {
```

Comment:
IMO the mode is a leaky abstraction of the Ptsotch interface and not optimal -> however, changing this might be annoying and not worth the effort -> you decide

Reply:
I know... :( I'll probably leave it for now like that—at least for now.